C_CPP/Dynamic Memory allocation/malloc,calloc,realloc.md

```diff
@@ -0,0 +1,149 @@
+# Dynamic Memory Allocation in C language
+
+The memory which is allocated using the concept of Static Memory Allocation is fixed and the size cannot be increased or decreased during run time. Hence, in order to tackle this problem, we use the concept of Dynamic Memory Allocation.
+
+As the name suggests, the method of allocating memory at the time of execution (during run time), is called Dynamic Memory Allocation. Here user has flexibility to allocate memory according to their needs.
+
+Heap is the segment of memory where dynamic memory allocation takes place. Heap, is basically an area of memory, where memory is allocated or deallocated randomly.
+
+The allocated memory can only be accessed through pointers.
+
+## Built-in Functions
++ malloc()
++ calloc()
++ realloc()
++ free()
+
+
+## malloc()
+
++ malloc is a built-in function which is declared under the header file <stdlib.h>
++ It is the abbreviation for "memory allocation".
++ It is used to dynamically allocate a single large block of contiguous memory according to the size specified by the user.
++ Syntax:
+
+ <pre> ``` int *q = (cast-type*)malloc(n * sizeof(cast-type)) ``` </pre>    where n is the size as mentioned by the user. 
```

Comment:
![image](https://user-images.githubusercontent.com/63473496/135729556-21593a46-713b-44d5-84aa-31243de82da8.png)

Backticks `  shouldn't appear right?

Reply:
@Suvraneel I actually wanted them to stay there originally, just for beautification purposes, but I've removed them now. Pls merge.